common/initializer.go

```diff
@@ -485,6 +503,12 @@ func (i *Initializer) autoRetryLoop(ctx context.Context) {
 		return
 	}
 
+	if i.State() == StateFatal {
+		i.logger.Error().Msg("initialization fatal state, stopping auto-retry")
+		i.autoRetryActive.Store(false)
+		return
+	}
```

Comment:
We should avoid retrying a specific task not the whole initializer (e.g. 2 upstreams are fatal but 5 upstreams have transient errors)